api/views.py

```diff
@@ -85,30 +85,64 @@ def get_queryset(self):
 
         Show children that have 1) responded to studies you can view and 2) are your own children
         """
-        original_queryset = super().get_queryset()
+        children_for_active_users = super().get_queryset()
         # Users with can_read_all_user_data permissions can view all children/demographics of active users via the API
         if self.request.user.has_perm("accounts.can_read_all_user_data"):
-            return original_queryset
-        qs_ids = original_queryset.values_list("id", flat=True)
+            return children_for_active_users
+
         studies = get_objects_for_user(
             self.request.user, "studies.can_view_study_responses"
         )
-        study_ids = studies.values_list("id", flat=True)
-        return qs_ids.model.objects.filter(
-            (Q(response__study__id__in=study_ids) | Q(user__id=self.request.user.id)),
-            (Q(id__in=qs_ids)),
-        ).distinct()
+
+        consented_responses = get_consented_responses_qs().filter(study__in=studies)
+
+        child_ids = consented_responses.values_list("child", flat=True)
+
+        return children_for_active_users.filter(
+            (Q(id__in=child_ids) | Q(user__id=self.request.user.id))
+        )
 
 
-class DemographicDataViewSet(ChildViewSet):
+class DemographicDataViewSet(FilterByUrlKwargsMixin, views.ModelViewSet):
     """
     Allows viewing a list of all demographic data you have permission to view as well as your own demographic data.
     """
 
     resource_name = "demographics"
-    queryset = DemographicData.objects.filter(user__is_active=True)
+    queryset = DemographicData.objects.filter(
+        user__is_active=True,
+    )
     serializer_class = DemographicDataSerializer
+    lookup_field = "uuid"
     filter_fields = [("user", "user")]
+    http_method_names = ["get", "head", "options"]
+    permission_classes = [IsAuthenticated]
+    filter_backends = (OrderingFilter,)
+
+    def get_queryset(self):
+        """Queryset getter override.
+
+        Largely duplicated from ChildViewSet but not completely, so we should duplicate before introducing the wrong
+        abstraction.
+
+        :return: The properly configured queryset.
+        """
+        demographics_for_active_users = super().get_queryset()
+
+        if self.request.user.has_perm("accounts.can_read_all_user_data"):
+            return demographics_for_active_users
+
+        studies = get_objects_for_user(
+            self.request.user, "studies.can_view_study_responses"
+        )
+
+        consented_responses = get_consented_responses_qs().filter(study__in=studies)
+
+        demographics_ids = consented_responses.values_list("demographic_snapshot", flat=True)
```

Comment:
Just a note for the future - we could legally/ethically give access to demographic snapshots associated with any users associated with consented_responses, i.e. researchers are allowed to see updates to demographic data that the family makes after participation. This behavior is more straightforward and in line with what researchers get through the experimenter UI though.